lib/mutant/mutator/node/regexp.rb

```diff
@@ -0,0 +1,104 @@
+module Mutant
+  class Mutator
+    class Node
+
+      # Generic mutator
+      module RegularExpression
+        class Parser
+          def self.parse(source)
+            Node.new(Regexp::Parser.parse(source, 'ruby/2.3'))
+          end
+
+          class Node < DelegateClass(Regexp::Expression::Base)
```

Comment:
Compared to the benefit its not. Having to work with mutable / class per type nodes is uglish like hell.

I've been there. Mutant initially was using the RBX AST thats heavily biased to one specific operation: Bytecode emitting. Not a generic AST useful for transformations.